test/routes/home_route_test.dart

```diff
@@ -133,6 +135,11 @@ void main() {
 
   testWidgets('home screen - shows when permissions are granted and not searching for tracks',
       (WidgetTester tester) async {
+    late AppWidgetChannelObserver appWidgetChannelObserver;
+    await tester.runAsync(() => tester.pump()); // Wait for widget events from old app start to process.
```

Comment:
Old app?

Reply:
I had a lot of problems getting these tests to work reliably. This is a workaround because due to our singleton architecture and because tests can end before all futures complete, we were getting events from previous tests on the `appWidgetChannelObserver`.

`old app start` is poorly worded, `other tests` might be better, I'll rephrase it.